lib/bundlex/toolchain/common/unix/os_deps.ex

```diff
@@ -110,12 +110,13 @@ defmodule Bundlex.Toolchain.Common.Unix.OSDeps do
   end
 
   defp resolve_os_dep_provider(name, {:precompiled, url, lib_names}, native) do
-    disabled_apps =
-      Application.get_env(:bundlex, :disable_precompiled_os_deps, [])
-      |> Keyword.get(:apps, [])
-      |> Bunch.listify()
+    is_precompiled_disabled =
+      case Application.get_env(:bundlex, :disable_precompiled_os_deps, false) do
+        bool when is_boolean(bool) -> bool
+        [apps: apps] -> native.app in Bunch.listify(apps)
```

Comment:
It is not mentioned in the `Bundlex.Project` moduledoc, that `apps` might be also just name of the app, eg. `[apps: :my_app]`